### DISCUSSION TITLE
Batocera Lite/Koriki Alpha for RG35XX Plus and RG35XX H - 20240205

### DISCUSSION THREAD

**Discussion by acmeplus** [rg35xx-cfw/rg35xx-cfw.github.io#100] [Announcements]
### Important: 

* This is a consolidated image for both RG35XX Plus and RG35XX H. You can install and actually swap from one system to another while keeping the same content.
* Because of the consolidation of both device features, this release cannot be installed as an update.
* There are still some performance issues with this release, while there are cores/emulators that are running fine, some others are not there yet. In general the build needs to be updated with optimizations, but there's still work to do.

### New install:
* Flash image to a new sdcard (batocera-h5-rg35xx-plus-39-20240505.img.gz)
* Insert sdcard into the RG35XX Plus/H
* Press power a few seconds to boot
* NOTE: The first time the system boots it will take a bit longer while the SDCARD is getting expanded and the initial SHARE folder content gets populated

### Update from a previous version (2024/01/25 or higher):
* Extract the boot.tar.xz file
* Mount the SDCARD on your computer
* Delete the file ``boot/batocera`` from the first partition of the SDCARD (BATOCERA)
* Copy its contents to the third partition of the SDCARD (boot-resource or BOOT)
* Insert the SDCARD in your RG35XX plus and boot

## Cores/Emulators Status:
* Working: amstrad, mame, atari2600, atari5200, atari7800,  wonderswan color, amiga(uae4arm), c64 (libretro x64), pico8, scummvm (core), colecovision (gearcoleco), vectrex, dosbox, pcengine, pcenginecd, g&w, megacd, nes, gb, snes, gbc, gba, openbor, sg-1000, mastersystem, megadrive, gamegear, spectrum, neogeo, neocd, ngpc, psx, psp, odyssey2, drastic, n64 (.zip with parallel libretro core, z64/n64 with mupen64plus), flycast (standalone and core), msx (fmsx core)
* Ports working (you need your own files): MalditaCastilla, Spelunky, Shovel Knight, VVVVVV, CurseOfIssyos, SpaceCadet Pinball
* Broken: scummvm standalone (unless using an external mouse), x86k (bios issue)

### 2024/02/05:
* Fixed bluetooth pairing. Use EmulationStation menu to pair (``Main Menu`` -> `` Controller & Bluetooth Settings`` -> Pair Bluetooth Pads Automatically``). Many known controllers can be paired with that option, otherwise use the ``Pair a Bluetooth device manually``
* Added potential fix for compatibility with WPA3 wifi routers (I don't have one, so please check and report back)

### 2024/01/26:
* Fixed an error with the proper es_input.cfg not containing the RG35XX H analogue input mappings
* Added suspend/resume mode (short press power button). 
* Added forced shutdown: press ``MENU`` + ``POWER`` keys 
* Fixed battery charge icon

### 2024/01/25:
* Consolidated RG35XX Plus and RG35XX H firmware
* Fixed “not connected” network notification (added missing coreutils timeout)
* Fixed issues with some cores not starting (e.g. GBA). This was indirectly caused by the network “not connected” issue
* Fixed retroachievements not being enabled in retroarch after enabled in ES
* Fixed network SHARE issues (BUILDROOT share should be vivible on your network shares once the console is connected to the network)
* Added boot progress bar. Thanks to Rafapardo (spirax_es) who contributed this needed feature
* Added moonlight. Refer to [batocera moonlight documentation](https://wiki.batocera.org/systems:moonlight) for setup information 
* Added xash3d (Half Life) port. Refer to batocera documentation for setup ([Xash3D half flife forums](https://forum.batocera.org/d/6667-xash3d-half-life))
* Included VVVVVV data as authorized by Terry Cavanagh (see the original repo from @JohnnyonFlame https://github.com/JohnnyonFlame/VVVVVV/releases/tag/v2.4-r1 with)
* Fixed controller mappings for ports (MalditaCastilla, Shovel Knight, Spelunky, etc). Note that you still need to provide your own port content and copy it over to the corresponding port folder in ``roms/ports``

### 2024/01/22:
* Fixed shutdown, reboot. Thanks @xonglebongle (adixal) for the hint
* Switched back to GCC compilation, all cores recompiled
* Fixed flycast standalone freezing before game starts
* Fixed Drastic (NDS) not starting
* Added workaround for Drastic control mapping. Add the file ``es_systems_nds.cfg`` to ``[SHARE]/system/configs/emulationstation/``. To use the stylus keep pressed ``R2`` to use DPAD for stylus and A to "click". 
* Confirmed ADB is working
* Fixed Retroarchievements from EmulationStation (you need to have retroarchievements.org account)
* Fixed scrapers 

### 2024/01/19:
* Fixed SHARE partition automatic expansion for single SDCard configuration
* Change SHARE format to expat
* Added additional card filesystem consistency checks for both internal and external SHARE partitions 
* Fixed SSH credentials, use root/linux as default
* Fixed flycast_libretro (dreamcast, Naomi, atomiswave) and made core selection available from emulationstation. If you applied the 2024/01/16 es_system_dreamcast.cfg workaround, remove it. 
* Flycast standalone is still broken, once you are in dreamcast, Naomi, or atomiswave section, press ``SELECT`` to open the ``VIEW OPTIONS`` menu, then go to ``ADVANCED SYSTEM OPTIONS``, and in ``EMULATOR`` choose ``LIBRETRO: FLYCAST``
* Initial bluetooth work (still not usable via frontend)
* Added some GLES3 cores/emulators: mupen64plus-next and Yabasanshiro (Saturn).
 
### 2024/01/16:
* Fixed audio click noise (not perfect but now very subtle)
* Fixed audio volume controls (volume can be controlled with the keys and is saved during reboots)
* Fixed audio not working on some emulators/cores
* Added brightness controls (level is saved across reboots)
* Added samba support. You can mount the SHARE partition as a network drive

### 2023/12/20:
* Fixed Overclock settings that were not correctly setup. In EmulationStation, change default setting in emulationstation settings
* mupen64 now working
* Added additional cores (gearcoleco, gem, race, vm, bk, gearbox, meteor, quicknes, cannonball, mame2003-xtreme, crocosd, supafaust, etc.). Note that not all cores are mapped in EmulationStation yet, in some cases you will need to open retroarch, and select the core manually to test.

## Things that don't work:
* Charging the console will power it on
* Bluetooth
* Some cores/Standalone emulators:
  * Amiga performance is not there yet
* HDMI (not tested)
* Sleep/Supend
* Batocera bezels (not tested)
* Menu language change (graphical themes get updated, but the menu remains in English)

## Things that work:
* Wifi
* Controls
* Many cores/emulators
* EmulationStation
* Simplemenu
* Brigthness control (Menu + Volume keys)
* ADB

## Additional features:
* Switching default launcher (ES/Batocera, Simplemenu/Koriki, or Retroarch):
  * Create a new file in ``[SHARE]/system/customlauncher`` with the text ``simplemenu`` in it.  
  * The next time the system starts will launch simplemenu instead of EmulationStation
  * Possible values for the custom launcher file are: ``simplemenu``, ``emulationstation``, and ``retroarch``

[![ko-fi](https://ko-fi.com/img/githubbutton_sm.svg)](https://ko-fi.com/A0A1J951S)

<hr /><em>This discussion was created from the release <a href='https://github.com/rg35xx-cfw/rg35xx-cfw.github.io/releases/tag/rg35xx_plus_h_batocera_lite_20240205'>Batocera Lite/Koriki Alpha for RG35XX Plus and RG35XX H - 20240205</a>.</em>

**Comment by angeld7** [comment]
I see Sleep/Suspend is listed in the things that don't work, but it looks like it was added in the 1/26 release. Is it still not working?

**Reply by acmeplus** [reply]
It's working, updated the release page to reflect that (note the text above does not automatically update when I update a release).

The only issue with suspend is that it does not have a configured auto power off feature right now, so if you leave the device in sleep for a long time it will be dead... but that's true for all mobile devices.

**Reply by angeld7** [reply]
Thanks for confirming! I'm not sure if I would like an auto power off, never had something like that on a handheld. Basic sleep as you described is fine for me.

**Reply by dopazz** [reply]
If I leave my RG35XX H in sleep mode for 8 hours it consumes about 50% of the battery.  Is that expected, or does sleep mode need to be sleeping a bit deeper?  If that is expected then auto power off would be a welcome feature.

**Comment by dopazz** [comment]
When I performed the update steps last time it wiped and recreated the Share partition.  Is that the expected behavior?  Will that happen if I update with this release?

**Reply by acmeplus** [reply]
Make sure you don't copy (or delete) the boot/autoresize file. That way it won't be recreated.

**Reply by dopazz** [reply]
Do you mean delete the autoresize file that is inside boot.tar under /boot before copying boot.tar to the SD card?

When I look at the SD card I see:

- Partition	Name	Label	File System
- 0	Unallocated Space (36MiB)
- 1	boot	unknown	unknown file system
- 2	env	unknown	unknown file system
- 3	boot-resource	BATOCERA	FAT32
- 4	userdata	SHARE	EXFAT
- 5	Unallocated Space (36MiB)

I deleted the 2.7 GB batocera file from /boot on the third partition, with name "boot-resource" and label "BATOCERA".

I dropped boot.tar into the root of the third partition with name "boot-resource" and label "BATOCERA".

The instructions say "Delete the file boot/batocera from the first partition of the SDCARD (BATOCERA)" but this is not the first partition when I look at the SD card.

The instructions also say "Copy its contents to the third partition of the SDCARD (boot-resource or BOOT)"
When you say "copy its contents" do you mean un-XYzip the file and put boot.tar on the root of the boot-resource partition?  or do you mean unzip the XY, and then unTAR the boot file to the third partition?

Also confusingly, I see "boot" and "boot-resource" as two different partitions but the instructions refer to them as if they are the same.

I'm manipulating the files on the SD card with a live boot Linux USB; I can't see these partitions in Windows nor is the boot partition visible as a SMB share.

**Reply by acmeplus** [reply]
Sorry, the only partition you need to use for the update is the 3rd partition, BATOCERA. From a regular user point of view that's the first partition when the card is inserted on a computer. Linux sees all the other partitions, so that's where the confusion is.

Basically for an update from a previous version (1/20 or newer) you just need to delete the boot/batocera file and replace that with the one from the boot.tar.xz.

**Reply by dopazz** [reply]
Thank you, and thanks for updating the update steps on the release.  The 2.7 GiB file in boot.tar.xy is "batocera.update"  Should it be renamed to just "batocera" when I copy it to the /boot folder?

Edit: I copied it to the /boot folder as "batocera.update" and booted the SD card in the RG35XX H, it worked as expected and renamed itself to just "batocera"

**Reply by acmeplus** [reply]
Yes. The automatic build uses that extension for updates but it’s not working yet. So just rename that file to batocera and drop it in boot/batocera after deleting the existing one.

**Comment by jdub850000** [comment]
I wanted to start from scratch so re-flashed the OS image and that seems to work but after adding a bunch of rooms to the share drive I can't seem to get the system to recognise. Is there a link with some more info?

**Reply by acmeplus** [reply]
Did you let the system expand the card first? You need to flash the card and boot at least once so the card is expanded and the folders are populated. After that happens you should be able to insert the card on your computer and add content, or just use the network share, adb or ssh to add content.

**Reply by jdub850000** [reply]
Yes, I rebooted and waited for a bit. Is there a way to know if the process is finished other than looking at the share directory?

**Reply by jdub850000** [reply]
Looks like I didn't kill the partition before the re-flash. Duh. I'm gonna try it again.

**Reply by jdub850000** [reply]
I've got the share partition, but no folder/files in there. Any way I can force/re-trigger the folder population?

**Reply by jdub850000** [reply]
All good - Was looking in the wrong place. Thanks for your work!!

**Comment by luxmenz** [comment]
Hello Acmplus,

I tried to see what was happening on batocera in relation to the hdmi output, and I found that xrandr is not installed, as on other versions of linux. I have not found how to install it on this distribution.
Similarly, the "batocera-resolution" command works, but as soon as you put in an argument, like "setOutput", it doesn't work.

Can you tell me if you think it would be compromised to use the hdmi output on Batocera with our machines (RG35XX plus/H) ?

**Reply by acmeplus** [reply]
No, the xrandr is not used in this version because we don't have the same graphics drivers as other batocera releases (*). I have HMDI working on my development environment, it just needs to be updated to be included in the HDMI detection scripts so it works transparently without needing to use the command line.

(*) The RG35XX-Plus/H don't use any X11, KMSDRM, or similar graphics drivers, but instead a graphics Mali driver running via a custom SDL2 library.

**Reply by luxmenz** [reply]
Great news.
I know I'm a bit pushy, but for me it's important to be able to play on my TV too. I hope you get it.

**Comment by PSERUM** [comment]
Hello Acmeplus,

The WPA3 wifi is still not working after testing it both on my router and my mobile hotspot.

**Reply by acmeplus** [reply]
Ok, I don’t have a wpa3 router to test. See if you can provide the logs from system/logs. Double check your ssid/password is not in those.

**Reply by PSERUM** [reply]
Here are all the files in the system/logs. I tried to connect my laptop hotspot which is WPA2 and it works again. But my mobile hotspot and router with WPA+WPA3 mix does not work.
[batocera.log](https://github.com/rg35xx-cfw/rg35xx-cfw.github.io/files/14172845/batocera.log)
[es_launch_stderr.log](https://github.com/rg35xx-cfw/rg35xx-cfw.github.io/files/14172846/es_launch_stderr.log)
[es_launch_stdout.log](https://github.com/rg35xx-cfw/rg35xx-cfw.github.io/files/14172847/es_launch_stdout.log)

**Reply by acmeplus** [reply]
The only information that is relevant on that log is:
```bash
PING 8.8.8.8 (8.8.8.8): 56 data bytes
64 bytes from 8.8.8.8: seq=0 ttl=116 time=122.720 ms
```
but that could be when you connected to your laptop hotspot. I'll try to investigate more and let you know once I have a version to test.

**Reply by acmeplus** [reply]
The good news is that I just found out that my travel router has WPA2/WPA3, so I'll use that to test

**Reply by acmeplus** [reply]
Just tried forcing my router to both WPA2/WPA3 and WPA3, and was able to connect and transfer data without issues. Not sure if the problem may be wifi 6 related, although that shouldn't be an issue if the device is able to connect using stock.

**Reply by PSERUM** [reply]
I've tried to use my mobile hotspot too which is wifi 2.4ghz/5ghz mixed and they won't connect too.

**Reply by acmeplus** [reply]
Which specific WPA setting do you have? and which wifi mode? (b/g/n/ax)?

**Reply by PSERUM** [reply]
802.11ax, WPA3, 5 GHz. I'm also reflashing Batocera to see if that helps, since I updated it from the previous version

**Reply by PSERUM** [reply]
Tried reflashing the os, still doesn't work. And when I check the system/log I only see batocera.log which is blank.

**Comment by Hypervisor64** [comment]
nice so , i guess once HDMI gets sorted then... this will be more or less feature complete , in an unfinished state of course but .... the hardware will be able to use all of its features. i am excited.

**Reply by luxmenz** [reply]
Totally agree with you

**Comment by soaremicheledavid** [comment]
**UPDATE :** 

My second sd card, were i have all the roms, got corrupted out of nowhere, lost all my files, did not have any problems with previous updates. Always make a backup of your roms/bios files!

**Performance isuess :** 

**ATOMISWAVE & NAOMI** games have bad performance, only emulator that run decent is libretro: flycast, standalone lags, but has major drops in FPS, the same happens to **Nintendo DS** games, stuttering  and sound issues, stilus dont work, games that need a touch to start the games, cant be started. **N64** run decent only on Mupen64Plus : RICE , emulator, but has glitches. Before this new update, **Nintendo VIRTUAL BOY**, worked great, not has sound issues, crackling noise. **AMIGA** games dont run well, they are not playable, **C64** has the same issues. Atari **JAGUAR**, it lags so much the device is freezing, stops working.

**System isuess :** 

**Suspending the system**, consume battery too much.
**Retroarch** , dosent save input configuration, tried to reset the config, flashed a clean image, nothing works.
**System Booting**, takes about 45 seconds to open the system, tested it about 4 times, same result.
**HDMI**, not working, Moonlight, not working.

SimpleMenu runs well, that's strange, even has improved performance, but bad buttons mapping, some non existing. 
Talking about SimpleMenu, always loved the simplemenu and the general use of it, so i decided to make a theme for Batocera that looks like simplemenu, but with Batocera functionality, soon will be available in-menu-themes : https://github.com/soaremicheledavid/ES-Theme-CarbonX 

**Thanks for your work for Batocera!**, wish to have better perfomance for cores, maybe update all the cores, in next img, add some cores that are missing even for existing system, to have more options, sometimes a core works better for a individual game, there are people that want older system to be emulated.

**Comment by paolopartipilo** [comment]
Hi, thanks for your work. I would like to report the fact that the Dreamcast emulator, both standalone and core, works very badly, the audio crackles and is very slow, furthermore Retroarch also has a problem, if I enter the menu to exit the game or change settings it freezes on the screen and I have to force reset to restart the console

**Comment by dopazz** [comment]
I am trying to pair a PS4 or PS5 controller.  My experience with PS4:

1. Press and hold PlayStation and Share buttons to enter pairing mode on controller.
2. Select "Pair Bluetooth Pads Automatically" in the "Controller & Bluetooth Settings" menu in Batocera.
3. Scanning Bluetooth "Searching for devices..." appears in top right corner.
4. Status changes to "Connecting Wireless Controller".
5. Lights on PS4 controller change from quick flash (pair mode) to solid lit (connected).
6. PS4 controller turns off after about 5 seconds.
7. Batocera "gives up" in the Bluetooth status in top right corner.

After this has happened there is an entry in the "Pair a Bluetooth Device Manually" and "Forget a Bluetooth Device" menu that I assume is the PS4 controller I just tried to connect.  I tried turning the controller on and selecting the device in the manual pair menu, but again it shuts off after a few seconds.

I tried this several times, and one time Batocera displayed "Trusing Wireless Controller" which seemed further than it had gotten previously, but then the controller turned off and it gave up.

If I use a USB cable and plug this same PS4 controller into the RG35XX H it works as a USB controller as expected, just can't get it to connect over Bluetooth.

**Comment by charlessouzanetto** [comment]
I have the same issue with the previous version about nds games. They are mapped wrong and dont save the new configuration.

**Comment by alojzjakob** [comment]
1. still having issues with boot time, looks like it slows to a crawl and takes a long time to pass 10% -15% mark on loading bar. Any idea what it could be at that point of loading process? Is it my 120 gigs of games on 2nd card? :)
3. would be nice if it can remember brightness / volume level between power cycles

and once again, thanks a million for all your hard work!

**Comment by Minak-Kru** [comment]


**Reply by alojzjakob** [reply]
while selecting a game just press and hold B button to bring up the menu :)
from there you can favorite it or put in custom list, view game metadata, manual, scrape only that one game etc.

**Comment by acmeplus** [comment]
New Beta Release is now out and should fix some of the bugs discussed here: https://github.com/rg35xx-cfw/rg35xx-cfw.github.io/releases/tag/rg35xx_plus_h_batocera_lite_20240208

**Comment by renewedgiant** [comment]
can anyone help me get psx working i feel as if i have tried everything

**Comment by darrenyorston** [comment]
Has there been any progress in getting HDMI out to work? When I plug in my device nothing happens. Works fine on the default device OS so I know it isnt a hardware issue.
